octodns/provider/googlecloud.py

```diff
@@ -0,0 +1,343 @@
+            'values': gcloud_record.rrdatas
+        }
+
+    _data_for_AAAA = _data_for_A
+
+    def _data_for_CAA(self, gcloud_record):
+        return {
+            'values': [{
+                'flags': v[0],
+                'tag': v[1],
+                'value': v[2]}
+                for v in [shlex.split(g) for g in gcloud_record.rrdatas]]}
+
+    def _data_for_CNAME(self, gcloud_record):
+        return {
+            'value': gcloud_record.rrdatas[0]
+        }
+
+    def _data_for_MX(self, gcloud_record):
+        return {'values': [{
+            "preference": v[0],
+            "exchange": v[1]}
+            for v in [shlex.split(g) for g in gcloud_record.rrdatas]]}
+
+    def _data_for_NAPTR(self, gcloud_record):
+        return {'values': [{
+            'order': v[0],
+            'preference': v[1],
+            'flags': v[2],
+            'service': v[3],
+            'regexp': v[4],
+            'replacement': v[5]}
+            for v in [shlex.split(g) for g in gcloud_record.rrdatas]]}
+
+    _data_for_NS = _data_for_A
+
+    _data_for_PTR = _data_for_CNAME
+
+    def _data_for_SPF(self, gcloud_record):
+        if len(gcloud_record.rrdatas) > 1:
+            return {
+                'values': gcloud_record.rrdatas}
+        return {
+            'value': gcloud_record.rrdatas[0]}
+
+    def _data_for_SRV(self, gcloud_record):
+        return {'values': [{
+            'priority': v[0],
+            'weight': v[1],
+            'port': v[2],
+            'target': v[3]}
+            for v in [shlex.split(g) for g in gcloud_record.rrdatas]]}
+
+    _data_for_TXT = _data_for_SPF
+
+    def _rrset_for_A(self, gcloud_zone, record):
+        return gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, record.values)
+
+    _rrset_for_AAAA = _rrset_for_A
+
+    def _rrset_for_CAA(self, gcloud_zone, record):
+        return gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [
+                '{flags} {tag} {value}'.format(**record.data['value'])])
+
+    def _rrset_for_CNAME(self, gcloud_zone, record):
+        return gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [record.value])
+
+    def _rrset_for_MX(self, gcloud_zone, record):
+        return gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [
+                '{preference} {exchange}'.format(**v.data)
+                for v in record.values])
+
+    def _rrset_for_NAPTR(self, gcloud_zone, record):
+        return gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [
+                '{order} {preference} "{flags}" "{service}" '
+                '"{regexp}" {replacement}'
+                .format(**v.data) for v in record.values])
+
+    _rrset_for_NS = _rrset_for_A
+
+    _rrset_for_PTR = _rrset_for_CNAME
+
+    def _rrset_for_SPF(self, gcloud_zone, record):
+        if 'values' in record.data:
```

Comment:
`data` is essentially a `to_json` type method so it probably shouldn't be used in these. They should instead get at the values property `record.values`.

Specifically for TXT you'll probably want to check out https://github.com/github/octodns/pull/115 and determine whether gcloud automatically handles TXT values > 255 chars or if the provider needs to do `chunked_values`.

Reply:
understood.

Reply:
Curious. No google cloud does not handle >255 chars for quoted text values, so it would have to use `chunked_values` too. I will have a look at this as soon as I have handled the other stuff. Should be just to replace `values` with `chunked_values` though.

Reply:
OK so how should I implement this one? I propose that if this change gets merged before #115 then I will update #115 with fix for googlecloud provider, and if the other way  around I rebase this PR with master, or if both gets merged I can open a new PR with the chunked_values. What do you suggest?

Reply:
I'm hoping to get https://github.com/github/octodns/pull/115 reviewed & shipped shortly. If it goes first you can pull the changes in. If not I'll make the change in that PR and ping you to :eye: and test there.